### PR TITLE
Add Dockerfile for a development purposes container

### PR DESCRIPTION
**Pull request by devguyio**
This introduces the following changes
    
- Add support for building a kroxylicious fat jar in the pom file
- Change root logger level to INFO
- Adding a multi-stage Dockerfile which runs a minimal JRE + the fat jar for the development purposes only

### CODE REVIEW

**Review by SamBarker — COMMENTED** [review]
I don't see the point of a fat jar when building a docker image and longer term I'd like to use Google Jib to build the image as a layer per dependency just seems like the right model to me (I think that's what Quarkus is doing as well).

Changes look fine to me but haven't tested so just commenting rather than approving

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+RUN mvn -q -ntp -Pdist -Dquick -B -pl krpc-code-gen,kroxylicious package
+
+# Final runtime stage
+FROM ${RUNTIME_IMAGE}
+
+WORKDIR /app
+
+COPY --from=build /build/kroxylicious/target/kroxylicious-1.0-SNAPSHOT-jar-with-dependencies.jar kroxylicious-1.0-SNAPSHOT.jar
+COPY --from=build /build/kroxylicious/example-proxy-config.yml proxy-config.yml
```
Comment (SamBarker) [inline Dockerfile:50]: Given the config file is specified in the run command why bother with the re-name here?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+RUN mvn -q -ntp -B -pl krpc-code-gen install
+
+RUN mvn -q -ntp -Pdist -Dquick -B -pl krpc-code-gen,kroxylicious package
+
+# Final runtime stage
+FROM ${RUNTIME_IMAGE}
+
+WORKDIR /app
+
+COPY --from=build /build/kroxylicious/target/kroxylicious-1.0-SNAPSHOT-jar-with-dependencies.jar kroxylicious-1.0-SNAPSHOT.jar
```
Comment (SamBarker) [inline Dockerfile:49]: What does the fat-jar add here that copying the slim jar + libs dir wouldn't? 

Using a fat jar invalidates the dependencies when ever the code changes thus making container builds slower (IIRC) perhaps that's mitigated by the multi stage build

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#
```
Comment (SamBarker) [inline Dockerfile:1]: 😢 was going to suggest `Dockerfile` -> `Containerfile` but seems like `Containerfile` is just as podman specific as `Dockerfile` is docker specific and they [don't want](https://github.com/docker/cli/issues/3251#issuecomment-1063338859) to support it

File: kroxylicious/src/main/resources/log4j2.properties [hunk 4]
```diff
@@ -11,7 +11,7 @@ appender.console.name = STDOUT
 appender.console.layout.type = PatternLayout
 appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss} <%t> %-5p %c:%L - %m%n
 
-rootLogger.level = WARN
+rootLogger.level = INFO
```
Comment (SamBarker) [inline kroxylicious/src/main/resources/log4j2.properties:14]: I'm in favour of this change but I think there is an argument that the container should probably have it's own logging configuration and possibly expose DEBUG by default.

**Review by SamBarker — APPROVED** [review]
LGTM

**Comment by devguyio** [comment]
So I'm gonna close this PR until we've an agreement on whether we go with the Dockerfile or JIB plugin.

**Pull request closed without merging**